### PR TITLE
Expand datamanager error.

### LINKED ISSUE #4406: 32 bit tests failing
The 32 bit tests are currently failing, due to a failed attempt to install the `cryptography` library from source. See e.g. https://dev.azure.com/sunpy/sunpy/_build/results?buildId=9154&view=logs&jobId=a7b3aa55-7d57-562f-3433-7f6b2d4252da&j=2920fa54-56c6-5dfd-d3d8-7b1b694d90cb&t=bba4d609-c264-5f9c-e072-020a8a3b573f

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #4406

### CODE REVIEW

File: sunpy/data/data_manager/cache.py [hunk 1]
```diff
@@ -155,18 +155,24 @@ def _download_and_hash(self, urls):
             Path, hash and URL of the file.
         """
         def download(url):
-            path = self._cache_dir / get_filename(urlopen(url), url)
+            # Disable SSL check for this download.
```
Comment (Cadair) [inline sunpy/data/data_manager/cache.py:158]: :eyes: for why?

Comment (nabobalis) [reply sunpy/data/data_manager/cache.py:158]: SSL error was causing urlopen to fail. 
But it fails to download the file regardless.

Comment (Cadair) [reply sunpy/data/data_manager/cache.py:158]: We shouldn't do this here anyway, as it's going to apply to all users of the cache in all situations?

Comment (nabobalis) [reply sunpy/data/data_manager/cache.py]: Removed. I guess we can just hope the server recovers instead.

File: sunpy/data/data_manager/cache.py [hunk 2]
```diff
@@ -157,16 +157,17 @@ def _download_and_hash(self, urls):
...
             try:
                 return download(url)
             except Exception as e:
-                warn(e, SunpyUserWarning)
-                error = e
+                warn(f"{e}", SunpyUserWarning)
+                errors.append(f"{e}")
         else:
-            raise RuntimeError from error
+            raise RuntimeError(errors)
```
Comment (nabobalis) [inline sunpy/data/data_manager/cache.py:173]: None of this worked.

**Comment by dstansby** [comment]
What's the rest of this PR for that isn't just fixing the 32-bit tests? Is it done?

**Comment by nabobalis** [comment]
> What's the rest of this PR for that isn't just fixing the 32-bit tests? Is it done?

Just the error and yes.

**Pull request merged**